From 06f7c63155cecd11a2d15091576167356f054adc Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Thu, 16 Nov 2017 21:46:22 +0800
Subject: [PATCH 2809/5242] MLK-16833 ARM64: dts: freescale: imx8mq: enable
 over-drive mode by default

commit  9a1074ffacbed58f9df3d03da88d0a8bae90407e from
https://source.codeaurora.org/external/imx/linux-imx.git

As all the EVK boards except A0 support over-drive mode,
so enable it in default EVK board dtb and remove over
drive mode dtb.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 -
 .../arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts |   51 --------------------
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts   |   45 ++++++++---------
 3 files changed, 23 insertions(+), 74 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index ea5b5f9..edceff7 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -48,7 +48,6 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-dsi-adv7535.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-dsi-rm67191.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb \
-				 fsl-imx8mq-evk-od.dtb \
 				 fsl-imx8mq-evk-4k.dtb \
 				 fsl-imx8mq-evk-drm.dtb \
 				 fsl-imx8mq-evk-lcdif-dsi.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts
deleted file mode 100644
index 3722a3b..0000000
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts
+++ /dev/null
@@ -1,51 +0,0 @@
-/*
- * Copyright (C) 2016 Freescale Semiconductor, Inc.
- * Copyright 2017 NXP
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * as published by the Free Software Foundation; either version 2
- * of the License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- */
-
-#include "fsl-imx8mq-evk.dts"
-
-/ {
-	regulators {
-		reg_gpio_dvfs: regulator-gpio {
-			compatible = "regulator-gpio";
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_dvfs>;
-			regulator-min-microvolt = <900000>;
-			regulator-max-microvolt = <1000000>;
-			regulator-name = "gpio_dvfs";
-			regulator-type = "voltage";
-			gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
-			states = <900000 0x1 1000000 0x0>;
-		};
-	};
-
-	pwmleds {
-		status = "disabled";
-	};
-};
-
-&pwm2 {
-	status = "disabled";
-};
-
-&A53_0 {
-	operating-points = <
-		/* kHz    uV */
-		1500000 1000000
-		1300000 1000000
-		1000000 900000
-		800000  900000
-	>;
-	dc-supply = <&reg_gpio_dvfs>;
-};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 8b5151e..41bc2e2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -39,6 +39,18 @@
 			gpio = <&gpio2 19 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_gpio_dvfs: regulator-gpio {
+			compatible = "regulator-gpio";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_dvfs>;
+			regulator-min-microvolt = <900000>;
+			regulator-max-microvolt = <1000000>;
+			regulator-name = "gpio_dvfs";
+			regulator-type = "voltage";
+			gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+			states = <900000 0x1 1000000 0x0>;
+		};
 	};
 
 	modem_reset: modem-reset {
@@ -80,16 +92,6 @@
 		spdif-out;
 		spdif-in;
 	};
-
-	pwmleds {
-		compatible = "pwm-leds";
-
-		ledpwm2 {
-			label = "PWM2";
-			pwms = <&pwm2 0 50000>;
-			max-brightness = <255>;
-		};
-	};
 };
 
 &clk {
@@ -152,12 +154,6 @@
 			>;
 		};
 
-		pinctrl_pwm2: pwm2grp {
-			fsl,pins = <
-				MX8MQ_IOMUXC_GPIO1_IO13_PWM2_OUT	0x16
-			>;
-		};
-
 		pinctrl_dvfs: dvfsgrp {
 			fsl,pins = <
 				MX8MQ_IOMUXC_GPIO1_IO13_GPIO1_IO13	0x16
@@ -481,12 +477,6 @@
 	status = "okay";
 };
 
-&pwm2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm2>;
-	status = "okay";
-};
-
 &uart1 { /* console */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
@@ -655,3 +645,14 @@
 	reg = <0x0 0xb8000000 0x0 0x10000>;
 	status = "okay";
 };
+
+&A53_0 {
+	operating-points = <
+		/* kHz    uV */
+		1500000 1000000
+		1300000 1000000
+		1000000 900000
+		800000  900000
+	>;
+	dc-supply = <&reg_gpio_dvfs>;
+};
-- 
1.7.9.5

